### DISCUSSION TITLE
`cmake.buildDirectory` not used

### DISCUSSION THREAD

**Discussion by bradenmcd** [microsoft/vscode-cmake-tools#4053] [Q&A] [answered]
Why might the value for `cmake.buildDirectory` not be used?

The entirety of my user `settings.json` is this:
```json
{
    "window.autoDetectColorScheme": true,
    "cmake.showOptionsMovedNotification": false,
    "cmake.pinnedCommands": [
        "workbench.action.tasks.configureTaskRunner",
        "workbench.action.tasks.runTask"
    ],
    "cmake.buildDirectory": "${userHome}/build/${workspaceRootFolderName}"
}
```

There is no project-specific `settings.json` (i.e., no `.vscode` directory for the project); yet, the `cmake` command line that VS Code executes looks like this:
```
[proc] Executing command: "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DCMAKE_BUILD_TYPE=RelWithDebInfo -SC:/Users/Braden/OneDrive/Source/Repos/hello-cpp -BC:/Users/Braden/OneDrive/Source/Repos/hello-cpp/out/build/windows-x64-v143-RelWithDebInfo -G Ninja
```

Why might the `cmake.buildDirectory` setting not be picked up?

I'm using version 1.19.51 of the CMake Tools.

**Comment by gcampbell-msft** [comment, marked as answer]
@bradenmcd Without knowing what your project looks like, it's hard to say, but my best guess would be that if you are using CMakePresets.json, then we don't use the `cmake.buildDirectory`, we use the settings from your CMakePresets.
